### ISSUE TITLE
min_length rule should fail when field is empty

### ISSUE THREAD

**Issue by lsfratel** [mako-framework/framework#318]
###  Description
---
The validation rule min_length not failing when field is as empty string. It only fails when the string contains at least one character.
...

###  Steps to reproduce
---
```php
private $rules = [
  'name' => ['string', 'min_length(4)', 'max_length(255)'],
];

#[Middleware(InputValidation::class)]
public function create() {
  $body = $this->request->getBody();
  $data = $this->getValidatedInput($body->all(), $this->rules);
  
  return $this->jsonResponse(compact('data'));
}
```
If name is sent as empty string, it pass the validation, if name is sent with at least one character, the validation fails.

btw: NotEmpty rule is not registered.
...

### Versions & Platform
---

| What         | Version                  |
|--------------|--------------------------|
| PHP Version  | 8.2.12                    |
| Mako Version | 9.1.3                    |
| OS           | debian bookworm                        |
| Database     | postgres 16.0                        |

**Comment by lsfratel** [comment]
> I belive this is because validation rules aren't executed on empty fields unless they are marked as required.

but if is in the body and not null should be validated

**Comment by freost** [comment]
That would be a massive backwards compatibility break for traditional HTML forms since empty fields are submitted as empty strings but I could add a flag that overrides the default functionality.

Something like this:
```
$this->getValidatedInput($body->all(), $this->rules, validateEmptyFields: true);
```

**Comment by freost** [comment]
I have fixed the `not_empty` rule registration in 9.1.4 so if you add the rule to your rule-sets then the validation will fail if the field is included in the request body while being empty.

I have also added the `$validateEmptyFields` parameter to the upcoming 10.0.0 release.
